### PR TITLE
fix(home-manager/gtk): support all tweaks

### PR DESCRIPTION
**Pull request by isabelroses**
gtk will be the death of me

### CODE REVIEW

**Comment by isabelroses** [comment]
Will need changes after https://github.com/NixOS/nixpkgs/pull/324265. Worth waiting till its merged and backported.

File: modules/home-manager/gtk.nix [hunk 1]
```diff
@@ -121,7 +122,7 @@ in
     (mkIf enable {
       gtk.theme =
         let
-          gtkTweaks = concatStringsSep "," cfg.tweaks;
+          gtkTweaks = if cfg.tweaks == [ ] then "default" else (concatStringsSep "," cfg.tweaks);
         in
         {
           name = "catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}+${gtkTweaks}";
```
Comment (eljamm) [inline modules/home-manager/gtk.nix:128]: Following the changes from the nixpkgs PR, if tweaks is an empty list, the name will become:

```
"catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}"
```

In which case we'd need to do something like this:

```suggestion
          gtkTweaks = if cfg.tweaks == [ ] then "" else (concatStrings [ "+" (concatStringsSep "," cfg.tweaks) ]);
        in
        {
          name = "catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}${gtkTweaks}";
```

File: modules/home-manager/gtk.nix [hunk 2]
```diff
@@ -121,10 +122,10 @@ in
     (mkIf enable {
       gtk.theme =
         let
-          gtkTweaks = concatStringsSep "," cfg.tweaks;
+          gtkTweaks = if cfg.tweaks == [ ] then "" else (concatStrings [ "+" (concatStringsSep "," cfg.tweaks) ]);
```
Comment (eljamm) [inline modules/home-manager/gtk.nix:125]: Apologies, I didn't format the suggestion...

```suggestion
          gtkTweaks =
            if cfg.tweaks == [ ] then
              ""
            else
              (concatStrings [
                "+"
                (concatStringsSep "," cfg.tweaks)
              ]);
```

**Comment by isabelroses** [comment]
Both PRs hit their respective branches, so we should be good to merge now.

**Review by eljamm — APPROVED** [review]

File: modules/home-manager/gtk.nix [hunk 3]
```diff
@@ -121,10 +123,17 @@ in
...
         let
-          gtkTweaks = concatStringsSep "," cfg.tweaks;
+          gtkTweaks =
+            if cfg.tweaks == [ ] then
+              ""
+            else
+              (concatStrings [
+                "+"
+                (concatStringsSep "," cfg.tweaks)
+              ]);
```
Comment (getchoo) [inline modules/home-manager/gtk.nix:133]:
```suggestion
          gtkTweaks = "+" + concatStringsSep "," cfg.tweaks;
```
there are like a thousand different ways you could do this, but i just found this to be a bit cleaner

File: modules/home-manager/gtk.nix [hunk 4]
```diff
@@ -121,10 +123,17 @@ in
...
         in
         {
-          name = "catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}+${gtkTweaks}";
+          name = "catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}${gtkTweaks}";
```
Comment (getchoo) [inline modules/home-manager/gtk.nix:136]:
```suggestion
          name = "catppuccin-${cfg.flavor}-${cfg.accent}-${cfg.size}" + lib.optionalString (cfg.tweaks != [ ]) gtkTweaks;
```

File: modules/home-manager/gtk.nix [hunk 5]
```diff
@@ -6,6 +6,7 @@
 }:
 let
   inherit (lib)
+    concatStrings
```
Comment (getchoo) [inline modules/home-manager/gtk.nix:9]:
```suggestion
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: modules/home-manager/gtk.nix (hunk 3 above)
Comment (eljamm) [reply modules/home-manager/gtk.nix:133]: Indeed, that's much better.

**Review by getchoo — APPROVED** [review]
thanks! hopefully this will be the last gtk PR now...

**Pull request merged**